### DISCUSSION TITLE
compile code that uses workers

### DISCUSSION THREAD

**Discussion by mkvlrn-cm42** [oven-sh/bun#7846] [Q&A] [answered]
TL;DR: how do you compile code that relies on the use of workers, which are not direclty imported in code, but rather referenced in a new instance of `Worker`?

`bun build --entrypoints ./app.ts ./workers/workerA.ts ./workers/workerB.ts --outdir ./build` followed by `bun run build/app.js` works flawlessly.

`bun build --entrypoints ./app.ts ./workers/workerA.ts ./workers/workerB.ts --compile --outfile app` fails because `error: multiple entry points are not supported with --compile`.

[the docs](https://bun.sh/docs/bundler/executables#embedding-files) say that I have to import files in my code to have them in the compiled file, but to use workers I don't import them, I just instantiate a `Worker` pointing to that file.

by doing that, in a regular compilation `bund build app.ts --compile --outfile app` and by running the resulting executable, I get an error when the Worker is to be initialized in code, stating that the module was not found.

and by just importing the workers at the top of the app to have them in the built bundle, I get the same error as before, not being able to find the worker module.

**Comment by Jarred-Sumner** [comment, marked as answer]
this is a great question

can you file an issue? We can make `new Worker` create a new entry point dynamically, sort of similar to import expressions

**Reply by mkvlrn-cm42** [reply]
[done](https://github.com/oven-sh/bun/issues/7901#issue-2058961498), thank you
